src/schema/me/create_credit_card_mutation.js

```diff
@@ -58,5 +99,14 @@ export default mutationWithClientMutationId({
     }
 
     return createCreditCardLoader({ token, provider: "stripe" })
+      .then(result => result)
+      .catch(error => {
+        const formattedErr = formatGravityError(error)
+        if (formattedErr) {
+          return { ...formattedErr, _type: "GravityMutationError" }
+        } else {
+          return new Error(error)
```

Comment:
Is the intent here to fallback to the top-level GraphQL error handling? If so, this should I think re-throw instead of return.

Reply:
I updated both instances to `throw`, although poking around the codebase it seems that we're using `return` and `throw` interchangeably in these situations (which suggests that both work). If we prefer one over the other we should probably follow up to standardize.

Reply:
It depends on what the intent is. If we want a returned Promise object to remain rejected, and thus leave error handling up to others, then we should re-throw. Returning rather than re-throwing will make the returned Promise object be in a resolved state and thus upstream error handling won’t be triggered.

Reply:
```
node
> const rejected = new Promise((_, reject) => reject(new Error("oh noes"))).catch(err => { throw err })
undefined
> (node:12719) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: oh noes
(node:12719) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

> const resolved = new Promise((_, reject) => reject(new Error("oh noes"))).catch(err => { return err })
undefined
> rejected
Promise {
  <rejected> Error: oh noes
    at Promise (repl:1:52)
    at Promise (<anonymous>)
    at repl:1:18
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:120:20)
    at REPLServer.emit (events.js:210:7),
  domain: 
   Domain {
     domain: null,
     _events: { error: [Function: debugDomainError] },
     _eventsCount: 1,
     _maxListeners: undefined,
     members: [] } }
> resolved
Promise {
  Error: oh noes
    at Promise (repl:1:52)
    at Promise (<anonymous>)
    at repl:1:18
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:120:20)
    at REPLServer.emit (events.js:210:7),
  domain: 
   Domain {
     domain: null,
     _events: { error: [Function: debugDomainError] },
     _eventsCount: 1,
     _maxListeners: undefined,
     members: [] } }
```